### ISSUE TITLE
change in instance name for openstack with linchpin 1.6.3

### ISSUE THREAD

**Issue by bhavikbhavsar** [CentOS-PaaS-SIG/linchpin#737]
In linchpin version 1.0.4 that we were using,
the openstack instance name would form based on below tasks in provision_os_server.yml

```
- name: "provisioning os_server resources"
  os_server2:
    state: "{{ state }}"
    name: "{{ res_grp_name }}_{{ res_def['res_name'] | default(res_def['name']) }}"
```

This seems to have changed in latest version 1.6.3

```
 name: "provision/teardown os_server resources with provided auth"
  os_server:
    state: "{{ state }}"
    auth: "{{ auth_var }}"
    name: "{{ prov_inst }}"

```
So now when instance is provisioned with latest version 1.6.3

For topo file as below:

```
resource_groups:
- credentials:
    filename: clouds.yaml
    profile: devstack
  resource_definitions:
  - count: 1
    fip_pool: 10.8.240.0
    flavor: m1.large
    image: rhel-7.5
    keypair: idm-jenkins
    name: master
    networks:
    - idm-jenkins
    type: os_server
  resource_group_name: linchpin_768
  resource_group_type: openstack
topology_name: openstack_topo
```
The instance name it assigns is `master1`  which was before `linchpin_768_master1`.

Is there any specific reason for this change?

My use case:
I had an working automation for my ci to provision instances with different name each time it runs `linchpin up` to automatically append 3 digit random number to the resource_group_name.

**Comment by samvarankashyap** [comment]
@bhavikbhavsar @herlo : By default linchpin generates unique id for resources . However, I believe there is a option in linchpin.conf to disable unique id generation . 
@herlo : I was checking the linchpin.conf in examples workspace I could not find the option . Do you know where the option is ?

**Comment by samvarankashyap** [comment]
@bhavikbhavsar : Got the option . 
You can disable the above behaviour by setting in linchpin.conf 
```
enable_uhash = False
```

**Comment by bhavikbhavsar** [comment]
@samvarankashyap I changed the parameter you mentioned here:
/usr/lib/python2.7/site-packages/linchpin/templates/linchpin.conf


Still it assigned master1 as the instance name. 

My only requirement is to get a different instance name each time I run `linchpin up`.

**Comment by samvarankashyap** [comment]
As i mentiomed in previous comments , By default linchpin assigns unique id . Which inturn creates a new instance everytime you provision . 
Further, 
You should not change it in the templates path as mentioned above . You should be changing linchpin.conf directory where linchpin command runs ie., Your workspace. ( to disable the unique id ).

**Comment by herlo** [comment]
One clarification, linchpin.conf sets the enable_uhash = False by default. In our workspaces, we set 'enable_uhash = True' because it provides a bunch of functionality.

**Comment by samvarankashyap** [comment]
I apologize for confusion as said by herlo its false by default

**Comment by bhavikbhavsar** [comment]
@samvarankashyap np, I wanted to set `enable_uhash = true` actually, now found the linchpin.conf.

I was using the an older linchpin workspace (created with 1.0.4 version) with latest linchpin version
so it wasn't having the linchpin.conf file.

I created a new workspace with `linchpin init` and changed the `enable_uhash = true` and now able to set unique instance name each time thanks for the help. 

Closing this issue.

**Comment by bhavikbhavsar** [comment]
@samvarankashyap np, I wanted to set `enable_uhash = true` actually, now found the linchpin.conf.

I was using the an older linchpin workspace (created with 1.0.4 version) with latest linchpin version
so it wasn't having the linchpin.conf file.

I created a new workspace with `linchpin init` and changed the `enable_uhash = true` and now able to set unique instance name each time thanks for the help. 

Closing this issue.
